script.js

```diff
@@ -1,4 +1,108 @@
+    if (
+      (input == "scissors" && computerChoice == "stone") ||
+      (input == "paper" && computerChoice == "scissors") ||
+      (input == "stone" && computerChoice == "paper")
+    ) {
+      // Update value of global variable for computer win count
+      compWin = compWin + 1;
+      // Create variable which outputs user win percentage
+      winPercentage = (userWin / (userWin + compWin + draws)) * 100;
+      var displayMessage = toDetermineMessage();
+      myOutputValue = `Sorry ${userName}, you lost! 😿 <br> You chose ${input} <br>  The computer chose ${computerChoice} <br> You've won: ${userWin} times. <br> The computer has won ${compWin} times. <br> You have drawn: ${draws} times. <br> Your win percentage is ${winPercentage}%. <br> ${displayMessage} <br> <br>Please try again! ✂️🧻🗿`;
+    }
+
+    // Create condition where scissors beats paper and user wins
+    // Create condition where paper beats stone and user wins
+    // Create condition where stone beats scissors and user wins
+    if (
+      (input == "scissors" && computerChoice == "paper") ||
+      (input == "paper" && computerChoice == "stone") ||
+      (input == "stone" && computerChoice == "scissors")
+    ) {
+      // Update value of global variable for user win count
+      userWin = userWin + 1;
+      // ?Call win percentage variable into winning conditional
+      winPercentage = (userWin / (userWin + compWin + draws)) * 100;
+      displayMessage = toDetermineMessage();
+      myOutputValue = `Congratulations ${userName}, you won! 😁 <br> You chose ${input} <br> The computer chose ${computerChoice}. <br> You've won: ${userWin} times. <br> The computer has won ${compWin} times. <br> You have drawn: ${draws} times. <br> Your win percentage is ${winPercentage}%. <br> ${displayMessage}  <br> <br>Wanna try again? ✂️🧻🗿`;
+    }
+    // Create condition where input = program output and result is a draw
+    if (input == computerChoice) {
+      // Update value of global variable for draws count
+      draws = draws + 1;
+      // ?Call win percentage variable into winning conditional
+      winPercentage = (userWin / (userWin + compWin + draws)) * 100;
+      displayMessage = toDetermineMessage();
+      myOutputValue = `Darn ${userName}, it's a draw! 😡 <br> You chose ${input} <br> The computer chose the same! <br> You've won: ${userWin} times. <br> The computer has won ${compWin} times. <br> You have drawn: ${draws} times. <br> Your win percentage is ${winPercentage}%. <br> ${displayMessage} <br><br> Please try again! ✂️🧻🗿`;
+    }
+    // If user types anything other than "scissors", "paper" or "stone", inform them that only these 3 words are vaild options
+    if (input != "scissors" && input != "paper" && input != "stone") {
+      myOutputValue = `Sorry ${userName}, you entered an invalid option! 😅 <br> Please enter one of the following options: <br> <br> ✂️ scissors  <br><br>🧻 paper  <br><br> 🗿 stone `;
+    }
+  }
```

Comment:
this section is quite logic intensive and is only applicable when the user is in SPS mode. Perhaps you can extract this part into a helper function that is called when the user chooses to play SPS. This will make your main function shorter and easier to read, where I won't have to read how each mode is executed to understand the general flow of the game. This will also make it easier for you to read and debug your code when working with more complex code in the future. A general rule of thumb is that if you see your function spanning more than 3/4 of your screen, you should consider if you could extract some logic out into a helper function.

Reply:
Thank you for the advice. Will work on refactoring my code in the subsequent sections and projects. I did try it once for this section but kept getting errors and got quite confused and overwhelmed because I had to keep jumping around. But I'll give it another try.